prime-router/src/main/kotlin/fhirengine/translation/hl7/utils/FhirPathUtils.kt

```diff
@@ -60,6 +60,16 @@ object FhirPathUtils : Logging {
         }
     }
 
+    /**
+     * Is the provided path a valid FHIR path given the evaluation context?
+     */
+    fun validatePath(path: String, evaluationContext: FHIRPathEngine.IEvaluationContext): Boolean {
+        return runCatching {
+            pathEngine.hostServices = evaluationContext
```

Comment:
I'm not a huge fan of mutating `pathEngine.hostServices` in the `FhirPathUtils` object before every call. It could leave it in a bad state. It also looks like its used outside the object in `ProcessFhirCommands` which could cause some undefined behavior.

Reply:
What about keeping the existing hostServices in a temp variable then resetting it afterwards?

Though this would only happen when running a `validate-yaml` command from the cli right?

Depending on above, maybe this warrants adding a TODO or a tech-debt ticket for a temporary context change.

Reply:
It's safe currently because the calls to mutate it happen during cli commands which runs a fresh version of the app every time.

The code called during the pipeline always resets the state to what it needs through appContext. It's not a danger yet but it is definitely a code-smell that could lead to a weird bug in the future. I'll leave a comment to think about how we could fix it.

Reply:
Reminder to leave comment

Reply:
Yeah I figured this wouldn't persist. I still think it would be a good idea to temporarily `hostServices` and then set it back after the function call.

Reply:
Done.